src/runtime_src/core/pcie/common/system_pcie.cpp

```diff
@@ -45,4 +45,40 @@ get_devices(boost::property_tree::ptree& pt) const
   pt.add_child("devices", pt_devices);
 }
 
+void
+system_pcie::
+bdf2index(uint16_t& index, const std::string& bdfStr) const
+{
+  // Extract bdf from bdfStr.
+  int dom = 0, b= 0, d = 0, f = 0;
+  index = std::numeric_limits<uint16_t>::max();
+  char dummy;
+  std::stringstream s(bdfStr);
+  size_t n = std::count(bdfStr.begin(), bdfStr.end(), ':');
+  
+  if (n == 1)
+    s >> std::hex >> b >> dummy >> d >> dummy >> f;
+  else if (n == 2)
+    s >> std::hex >> dom >> dummy >> b >> dummy >> d >> dummy >> f;
+  if ((n != 1 && n != 2) || s.fail()) {
+    std::string errMsg = boost::str( boost::format("Can't extract BDF from '%s'") % bdfStr);
+    throw error(errMsg);
+	}
+
+  for (uint16_t i = 0; i < get_total_devices().first; i++) {
+    auto device = get_mgmtpf_device(i);
+    boost::any bus, dev, func;
+    device->query(xrt_core::device::QR_PCIE_BDF_BUS, typeid(b), bus);
```

Comment:
I suggest you short circuit if 'bus' doesn't match 'b', ditto for 'dev' and 'd', and 'func' and 'f'.